stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt

```diff
@@ -156,6 +160,17 @@ public class ChannelListViewModel(
 
     public data class State(val isLoading: Boolean, val channels: List<Channel>)
 
+    private fun parseMutedChannels(
+        channelsMap: List<Channel>,
+        channelMutesIds: List<String>?,
+    ): List<Channel> {
+        return channelsMap.map { channel ->
+            channel.copy().apply {
+                isMuted = channelMutesIds?.contains(channel.id) ?: false
```

Comment:
We can reuse our extension `Channel::isMutedFor`, can't we?

Reply:
We can, but in this case, we would need referential transparency because we would need the `User` class. Here we already have the list of muted channels.

Reply:
But we also have  a user value there. Look where this method is invoked

Reply:
Idea is simple. There is no sense having the same logic twice in our SDK. Here and as an extention

Reply:
This method is called twice, one of them with currentUser available, but the other is not... I can call it with `ChatDomain.instance().user.value` but then I would be inserting nullability in the method =|. I just saw `channelMutesIds: List<String>,` doesn't need to be nullable

Reply:
But... now taking a second look... maybe we don't need this at all:
```
stateMerger.addSource(queryChannelsController.mutedChannelIds.asLiveData())
```

Let me check this.

Reply:
No, we still need to listen for the `mutedChannelIds` changes.

Reply:
It would be also a bit odd that we listen for `mutedChannelIds`,  but we ignore its data... `User.channelMutes` can be out of sync with `QueryChannelsController.mutedChannelIds` for a super small period of time, but that would create a subtle bug in this method.